pkg/addons/registry/migrate/pod.go

```diff
@@ -0,0 +1,371 @@
+			},
+		},
+	}
+
+	pod.ObjectMeta.Labels = applyRegistryLabels(pod.ObjectMeta.Labels, dataMigrationPodName)
+
+	return pod
+}
+
+func ensureMigrationServiceAccount(ctx context.Context, cli client.Client) error {
+	newRole := rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "registry-data-migration-role",
+			Namespace: runtimeconfig.RegistryNamespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"apps"},
+				Resources: []string{"deployments"},
+				Verbs:     []string{"get", "list", "update"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"secrets"},
+				Verbs:     []string{"create"},
+			},
+		},
+	}
+	err := cli.Create(ctx, &newRole)
+	if client.IgnoreAlreadyExists(err) != nil {
+		return fmt.Errorf("create role: %w", err)
+	}
+
+	newServiceAccount := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: runtimeconfig.RegistryNamespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceAccount",
+			APIVersion: "v1",
+		},
+	}
+	err = cli.Create(ctx, &newServiceAccount)
+	if client.IgnoreAlreadyExists(err) != nil {
+		return fmt.Errorf("create service account: %w", err)
+	}
+
+	newRoleBinding := rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "registry-data-migration-rolebinding",
+			Namespace: runtimeconfig.RegistryNamespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     "registry-data-migration-role",
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      serviceAccountName,
+				Namespace: runtimeconfig.RegistryNamespace,
+			},
+		},
+	}
+
+	err = cli.Create(ctx, &newRoleBinding)
+	if client.IgnoreAlreadyExists(err) != nil {
+		return fmt.Errorf("create role binding: %w", err)
+	}
+
+	return nil
+}
+
+func ensureS3Secret(ctx context.Context, kcli client.Client) error {
+	accessKey, secretKey, err := seaweedfs.GetS3RWCreds(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("get seaweedfs s3 rw creds: %w", err)
+	}
+
+	obj := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Namespace: runtimeconfig.RegistryNamespace, Name: seaweedfsS3SecretName},
+		Data: map[string][]byte{
+			"s3AccessKey": []byte(accessKey),
+			"s3SecretKey": []byte(secretKey),
+		},
+	}
+
+	obj.ObjectMeta.Labels = seaweedfs.ApplyLabels(obj.ObjectMeta.Labels, "s3")
+
+	if err := kcli.Create(ctx, obj); client.IgnoreAlreadyExists(err) != nil {
+		return fmt.Errorf("create secret: %w", err)
+	}
+	return nil
+}
+
+func applyRegistryLabels(labels map[string]string, component string) map[string]string {
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	// TODO: why would we want to back this up?
```

Comment:
i don't think we should. i'd say get rid of these labels for the migration pod

Reply:
updated